### PR TITLE
Remove version pinnings for docs build

### PR DESCRIPTION
**Pull request by astrofrog**
If this works, this fixes https://github.com/astropy/astropy/issues/9147

The Matplotlib pinning can definitely go, the sphinx ones I'm not sure, but will try.

### CODE REVIEW

File: .readthedocs.yml [hunk 1]
```diff
@@ -13,7 +13,6 @@ python:
       extra_requirements:
         - docs
         - all
-    - requirements: docs/rtd_requirements.txt
```
Comment (pllim) [inline .readthedocs.yml]: We wouldn't know about this one until after merge, unless you want to try build RTD from your fork...

**Comment by bsipocz** [comment]
I very recently still run into the mpl one while using the mpl master branch, so wouldn't surprise if this is premature. 
Sphinx probably works though.

**Comment by astrofrog** [comment]
Matplotlib 3.1.2 has a fix for the issue and was released recently.

**Comment by astrofrog** [comment]
Testing the RTD build here: https://readthedocs.org/projects/astropy/builds/9887110/

**Comment by bsipocz** [comment]
The html build is still failing.

**Comment by pllim** [comment]
Does the Docker image need updating or something?

File: astropy/visualization/wcsaxes/core.py [hunk 2]
```diff
@@ -96,6 +96,8 @@ class WCSAxes(Axes):
     def __init__(self, fig, rect, wcs=None, transform=None, coord_meta=None,
                  transData=None, slices=None, frame_class=None,
                  **kwargs):
+        """
+        """
```
Comment (astrofrog) [inline astropy/visualization/wcsaxes/core.py]: @bsipocz @pllim this is what was causing the issue with Sphinx 2.2, the ``__init__`` docstring was being automatically inherited. Not a bug in Sphinx (in fact it was a bug that it wasn't inherited before)

File: examples/coordinates/plot_galactocentric-frame.py [hunk 3]
```diff
@@ -21,13 +21,13 @@
 Note that, for brevity, we may refer to the solar system barycenter as just "the
 sun" in the examples below.
 
--------------------
+---
```
Comment (pllim) [inline examples/coordinates/plot_galactocentric-frame.py]: Huh... What does this mean? Does `---` mean something special?

Comment (astrofrog) [reply examples/coordinates/plot_galactocentric-frame.py]: Hmm well it stopped the warnings about these headings but the output doesn't look good. Fixing...

Comment (astrofrog) [reply examples/coordinates/plot_galactocentric-frame.py]: I just went ahead and removed them. Basically Sphinx thinks they are heading lines and complains about them.

**Comment by astrofrog** [comment]
Travis has passed but somehow the badge didn't update

**Comment by bsipocz** [comment]
Thanks @astrofrog, good to have this fixed!

**Pull request merged**